File: Andriod/Practical/USL/app/src/main/java/com/example/usl/Utils/LocalRepositories.java
package com.example.usl.Utils;

import android.content.Context;

import androidx.preference.PreferenceManager;

import com.google.gson.Gson;

public class LocalRepositories {

    public synchronized static void saveAppUser(Context ctx, AppUser user) {

        String jsonString = new Gson().toJson(user);
        PreferenceManager.getDefaultSharedPreferences(ctx)
                .edit()
                .putString(Cv.PREFS_APP_USER, jsonString)
                .commit();


    }

    public synchronized static AppUser getAppUser(Context ctx) {

        String jsonString = PreferenceManager.getDefaultSharedPreferences(ctx)
                .getString(Cv.PREFS_APP_USER, "");

        return "".equals(jsonString) ?
                null : new Gson().fromJson(jsonString, AppUser.class);
    }

}

File: Andriod/Practical/usl_kotlin/app/src/main/java/com/usl/usl/utils/LocalRepositories.kt
package com.usl.usl.utils

import android.content.Context
import android.preference.PreferenceManager
import com.google.gson.Gson

@Suppress("DEPRECATION")
class LocalRepositories :Cv(){


    @Synchronized
    fun saveAppUser(ctx: Context?, user: AppUser?) {
        val jsonString = Gson().toJson(user)
        PreferenceManager.getDefaultSharedPreferences(ctx)
            .edit()
            .putString(super.PREFS_APP_USER, jsonString)
            .commit()
    }

    @Synchronized
    fun getAppUser(ctx: Context?): AppUser? {
        val jsonString = PreferenceManager.getDefaultSharedPreferences(ctx)
            .getString(super.PREFS_APP_USER, "")
        return if ("" == jsonString) null else Gson().fromJson(
            jsonString,
            AppUser::class.java
        )
    }

}